react/RuiExpandCollapseReact/src/RuiExpandCollapse.tsx

```diff
@@ -0,0 +1,84 @@
+/**
+ * Copyright Deloitte Digital 2019
+ *
+ * This source code is licensed under the BSD-3-Clause license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React, { createRef } from 'react';
+import '@rhythm-ui/expand-collapse';
+
+import {
+	IRuiExpandCollapseProps,
+	IHTMLRuiExpandCollapseElement,
+	IExpandCollapseDetailsProps,
+	IExpandCollapseSummaryProps,
+} from './IRuiExpandCollapse';
+
+export class RuiExpandCollapse extends React.Component<IRuiExpandCollapseProps> {
+	private ruiExpandCollapseEl = createRef<IHTMLRuiExpandCollapseElement>();
+
+	public static Details = class extends React.Component<IExpandCollapseDetailsProps> {
+		public render() {
+			const { as, ...otherProps } = this.props;
+
+			let elementType = 'div';
+			if (as) {
+				elementType = as;
+			}
+
+			const DetailsEl = React.createElement(elementType, {
```

Comment:
Is there a reason you've gone with createElement here? Could it be achieved like below?
```
const { as: Tag = 'div', ...otherProps } = this.props;
return (
  <Tag slot="details-content" {...otherProps />
);
```

Reply:
@jeffdowdle both ways work, I just jumped to using create element initially.  Checking out your version now is making typescript bark at me due to the use of `<Tag>`. I'm happy to spend some more time figuring out how to fix the TS error if you think it's worth using the above notation instead

Reply:
I'll leave it up to you, they both do the same thing :)